packages/babel-parser/src/parser/expression.js

```diff
@@ -707,7 +707,7 @@ export default class ExpressionParser extends LValParser {
   ): N.Expression {
     if (!noCalls && this.eat(tt.doubleColon)) {
       return this.parseBind(base, startPos, startLoc, noCalls, state);
-    } else if (this.match(tt.backQuote)) {
+    } else if (this.match(tt.templateMiddle) || this.match(tt.templateTail)) {
```

Comment:
Nit: since there are different places where we check `this.match(tt.templateMiddle) || this.match(tt.templateTail)`, we could add a `tokenIsTemplate` function.